main.py

```diff
@@ -27,13 +47,32 @@ def get_todays_gazette():
         "Content-Type": "application/json",
         "Authorization": os.getenv("MARIA_QUITERIA_API_TOKEN"),
     }
-
-    response = requests.get(url, headers=headers, params=params)
-    response_json = response.json()
-
-    for result in response_json["results"]:
-        gazettes.append(result)
-
+    success = False
+    number_of_attempts = 0
+    while success == False and number_of_attempts < 4:
+        try:
+            logger.info("Looking for gazettes")
+            response = requests.get(url, headers=headers, params=params)
+
+            response_json = response.json()
+            for result in response_json["results"]:
+                gazettes.append(result)
+            success = True
+        except KeyError as e:
+            logger.exception(e)
+            if (response.status_code in [401, 403]) and response.json()['code'] == "token_not_valid":
+                logger.info("Token is invalid or expired. Getting new token")
+                new_token = refresh_maria_quiteria_api_token()
+                headers = {
+                    "Content-Type": "application/json",
+                    "Authorization": new_token,
+                }
+            number_of_attempts += 1
+            logger.info(f"Trying to get gazettes again. Attempt {number_of_attempts}/3.")
+            continue
+    
+    if gazettes == []:
+        logger.debug("Failed to fetch gazettes")
```

Comment:
Aqui não é necessariamente uma falha. Haverão dias, como domingos, onde não haverão novos diários. Podemos remover o `if` e ter algo como:
```suggestion
    logger.info(f"Number of gazettes found: {len(gazettes)}")
```